### PR TITLE
User Agent for image factor (closes #64).

### LINKED ISSUE #64: How can I customize the http request header when requesting for external image?
Some external image restrict their response to only browser user-agent. For example,
http://localhost:8080/display?op=resize&url=http://i-give-you.pics/only/if/you/use/browser.jpg
this request will give you no image, with non-custom user-agent.

### PR DESCRIPTION
**Pull request by steevee**
`DefaultUserAgent` leaving room for overrides in future.
Options section of config seemed most logical place for `DefaultUserAgent`.
`HTTPStorage.UserAgent` property added to save passing user-agent around for every function.
`HTTPStorage` can be used when creating `http` `Src`and `Dst` storages, but as not involved in image factory `DefaultUserAgent` is not set.

### CODE REVIEW

File: config/constants.go [hunk 1]
```diff
@@ -6,6 +6,9 @@ const DefaultFormat = "png"
 // DefaultQuality is the default quality for processed images
 const DefaultQuality = 95
 
+// DefaultUserAgent is the default user-agent header to fetch images from URL with
+const DefaultUserAgent = ""
```
Comment (thoas) [inline config/constants.go]: we should add a default user agent with picfit and the current version

**Comment by thoas** [comment]
I have resolved conflicts :)

**Comment by thoas** [comment]
very nice work @steevee

File: config/constants.go (hunk 1 above)
Comment (steevee) [reply config/constants.go]: @thoas hrm... referencing `application.Version` within `config` package results in circular imports. 

In order of (least to most) kludgery, but how do you feel about:
1. moving `application.Version` into `config.Version`.
2. passing `application.Version` through `config.Load(path string, version string)` and eventually into `config.DefaultConfig(version string)` (could store separately on the `Config{}` struct for future reference too)
3. hardcoding `application.Version` value in `config.DefaultUserAgent`

Not really happy with any of these, so if you have a better idea, suggest away.

Comment (thoas) [reply config/constants.go]: Hmm.. tough situation let me think about it :)

Comment (steevee) [reply config/constants.go]: Another solution is to move constants into sub-packages, but would want to modify for all packages with them for consistency.

e.g.
```go
import (
    applicationConstants "github.com/thoas/picsfit/application/constants"
    configConstants "github.com/thoas/picsfit/config/constants"
    engineConstants "github.com/thoas/picsfit/engine/constants"
    imageConstants "github.com/thoas/picsfit/image/constants"
)
```

Comment (thoas) [reply config/constants.go]: let's move ``application/constants.go`` to ``constants/constants.go`` at the root, it will solve the issue I think

Comment (steevee) [reply config/constants.go]: OK done

**Comment by thoas** [comment]
thank you for your contribution @steevee 👍

**Pull request merged**